File: rust/src/pr005.rs
// Problem 5
// 
// 30 November 2001
// 
// 2520 is the smallest number that can be divided by each of the numbers
// from 1 to 10 without any remainder.
// 
// What is the smallest positive number that is evenly divisible by all of
// the numbers from 1 to 20?

use num::Integer;

define_problem!(pr005, 5, 232792560);

fn pr005() -> u64 {
    let mut accum = 1u64;

    for i in 2u64 .. 20 {
        accum = accum.lcm(&i);
        // accum = accum.lcm(&i);
        // accum = lcm(accum, i);
    }
    accum
}

File: swift/pr005.swift
// Problem 5
// 
// 30 November 2001
// 
// 2520 is the smallest number that can be divided by each of the numbers
// from 1 to 10 without any remainder.
// 
// What is the smallest positive number that is evenly divisible by all of
// the numbers from 1 to 20?
//
// 232792560

func pr005() -> Int64 {
    var n: Int64 = 1
    for j in Int64(2) ... 20 {
        n = lcm(n, j)
    }
    return n
}

func lcm(a: Int64, _ b: Int64) -> Int64 {
    return (a / gcd(a, b)) * b
}

func gcd(a: Int64, _ b: Int64) -> Int64 {
    var aa = a
    var bb = b
    while true {
        if bb == 0 {
            return aa
        }
        let tmp = aa%bb
        aa = bb
        bb = tmp
    }
}
